bigquery/google/cloud/bigquery/dataset.py

```diff
@@ -546,4 +545,91 @@ def table(self, table_id):
         :rtype: :class:`~google.cloud.bigquery.table.TableReference`
         :returns: a TableReference for a table in this dataset.
         """
-        return TableReference(self, table_id)
+        return TableReference(self.reference, table_id)
+
+
+class DatasetListItem(object):
+    """A read-only dataset resource from a list operation.
+
+    For performance reasons, the BigQuery API only includes some of the
+    dataset properties when listing datasets. Notably,
+    :attr:`~google.cloud.bigquery.dataset.Dataset.access_entries` is missing.
+
+    For a full list of the properties that the BigQuery API returns, see the
+    `REST documentation for datasets.list
+    <https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/list>`_.
+
+
+    Args:
+        resource (dict):
+            A dataset-like resource object from a dataset list response.
+    """
+
+    def __init__(self, resource):
+        self._properties = resource
+
+    @property
+    def project(self):
+        """Project bound to the dataset.
+
+        :rtype: str
+        :returns: the project.
+        """
+        return self._properties.get('datasetReference').get('projectId')
```

Comment:
Presumes that the `datasetReference` key is always populated appropriately (with at least an empty dict).

We've not been perfectly consistent about defending against "perverse" back-end responses, so maybe this one can go by.  In that case, is `self._properties['datasetReference']['projectId']` clearer?

If we *do* leave the `.get()` bits in place, then we need to exercise the implicit branches for missing `datasetReference` in the unit tests.

Reply:
I agree that `self._properties['datasetReference']['projectId']` is clearer in this case. I've added an explicit check for these properties in the constructor.